### ISSUE TITLE
Reset password broken when using Laravel's default users

### ISSUE THREAD

**Issue by underdpt** [laravel-filament/filament#601]
### Package

filament/filament

### Package Version

v1.10.12

### Laravel Version

v8.61.0

### Livewire Version

v2.6.5

### PHP Version

8.0.10

### Bug description

Once configured the app to use Laravel's default user table per the documentation, when trying to reset a user's password it fails with:

```
Illuminate\Database\QueryException
SQLSTATE[42S02]: Base table or view not found: 1146 Table '[table].filament_users' doesn't exist (SQL: select * from `filament_users` where `email` = [user@email] limit 1) 
```

I tracked it down to `src\Http\Livewire\Auth\RequestPassword.php` using `filament_users` broker to send the reset link (line 39):

```php
        $requestStatus = Password::broker('filament_users')->sendResetLink($this->validate());
```

This also happens on `src\Http\Livewire\Auth\ResetPassword.php` line 46.

This can be solved by changing that lines with something like:

```php
        $broker = config('filament.auth.guard') === 'filament' ? 'filament_users' : null;
        $requestStatus = Password::broker($broker)->sendResetLink($this->validate());
```

That way, we use `filament_users` table when using the default filament guard, and whatever is setup as the default otherwise.

For the request password to fully work we also need to add the function `sendPasswordResetNotification` to our `User` model from fillament's one, to be sure we're sending filament's signed url on the notification and not Laravel's one (which may also fail if we don't have the default Laravel's auth routes published). Maybe it's better to move that function to a trait and use that trait on the User model (filament's or custom one).

Anyway, this is missing on the documentation.

I can make a PR if you think this is the correct approach.

### Steps to reproduce

Configure filament to use the default Laravel User model and authentication. Create a new user and try to reset his password.

**Comment by danharrin** [comment]
Yes please, make a PR.

**Comment by underdpt** [comment]
I'm on it.

For the `sendPasswordResetNotification` issue on `User` model, do you prefer a new trait (and add it to the documentation) or maybe we can move that function to `src\Models\Concerns\IsFilamentUser.php`?

What would be your preffered approach?

**Comment by danharrin** [comment]
Can you use the existing trait please, thanks!

**Comment by danharrin** [comment]
Fixed by #606.
